### ISSUE TITLE
Is it possible to specify different cellxgene arguments for different dataset?

### ISSUE THREAD

**Issue by zeissmania** [Novartis/cellxgene-gateway#75]
Hi,
When we have multiple datasets under the same directory, sometimes, we need to use different cellxgene arguments according to the dataset (for example, --gene-sets-file)
I added the cellxgene_geneset variable in the env.py  file at line 17  
```
cellxgene_geneset = os.environ.get("CELLXGENE_GENESET", None)
```

And the value of the environment variable is like below
CELLXGENE_GENESET="h5ad_file1:geneset_path1@@h5ad_file2:geneset_path2@@h5ad_file3:geneset_path3 "

In the subprocess_backend.py file line 37,   I added the following 
```
if not cellxgene_geneset is None:
    try:
        gene_set_map = dict([_.split(':') for _ in cellxgene_geneset.strip().split('@@')])
    except:
        logging.getLogger("cellxgene_gateway").info(f"invalid CELLXGENE_GENESET argument: {cellxgene_geneset}")
    else:
        import os
        file_name = os.path.basename(file_path)
        gene_set_file = gene_set_map.get(file_name) or gene_set_map.get(file_path)
        if not gene_set_file is None and os.path.exists(gene_set_file):
            extra_args += f" --gene-sets-file {gene_set_file}"
```

I tested locally and it works.

**Comment by alokito** [comment]
Hi! We have implemented gene sets support a different way (see #81 ) However, if you want to do something different a very generic workaround is to set the CELLXGENE_LOCATION environment variable to a custom shell script or python program that modifies the launch arguments to cellxgene.

**Comment by alokito** [comment]
Just to give yout he TL;DR: As of v0.3.11, GATEWAY_ENABLE_ANNOTATIONS enables gene sets in a similar way to annotations.
